ui/src/timeMachine/components/builderCard/BuilderCardDropdownHeader.tsx

```diff
@@ -21,15 +21,30 @@ export default class BuilderCardDropdownHeader extends PureComponent<Props> {
 
   public render() {
     const {children, options, onSelect, selectedOption, testID} = this.props
+    const button = () => (
+      <Dropdown.Button status={ComponentStatus.Disabled} onClick={() => {}}>
+        {selectedOption}
+      </Dropdown.Button>
+    )
 
+    const menu = () => null
     return (
       <div className="builder-card--header" data-testid={testID}>
-        <SelectDropdown
-          options={options}
-          selectedOption={selectedOption}
-          testID="select-option-dropdown"
-          onSelect={onSelect ? onSelect : emptyFunction}
-        />
+        {options.length === 1 && (
```

Comment:
Two comments on this block:

Generally: if you're going to do a statement of the form
```
if condition then foo
if !condition then bar
```
it's typically more common, and easier to follow an `if/else`
```jsx
{options.length === 1 ? (<Dropdown />) : (<SelectDropdown />)}
```

Second: This method is inferring what state to put itself into based on, well an inference, rather than checking values or the state of the app.

The length of the options array being 1 _may_ indicate that we should only show the filter, but it may also indicate something else. The issue here isn't hiding group when we're in the checks overlay, it's hiding group when the options list has one item in it - we may only want to show group, but the inference is that if there's only a single item, it must show filter.

What we're interested in is whether this is happening in the checks overlay, right? So how about passing `isInCheckOverlay` as a prop down to `BuilderCardDropdownHeader` - if the prop is present and true, only show `filter`, otherwise show `group` and `filter`.

I know it seems persnickety, but these small things add up in a codebase. I prefer this because it moves the logic into a single place, and it makes it clear, in context, what the intent is - you don't have to look through two files to see what's happening. It also concentrates the logic - rather than spreading the logic to be implicit in the length of the array, which is set based on whether we're in the check overlay, it instead renders a different control based on whether we're in the check overlay. It's like a 'one hop' logic resolution, versus a 'two hop' one with the options array.